### PR TITLE
Upgrade to Emotion 11

### PR DESCRIPTION
**Pull request by joneubank**
**Description of changes**

These changes update emotion to version 11, and fix all the corresponding TypeScript type issues related to this update.

Actions taken include:
- replace `emotion-theming`, `@emotion/core`, and `@emotion/css` with `@emotion/react`
- refactor styled components with props from `styled<type, props>(type)` to `styled(type)<props>`
- added `PropsWithChildren` React type wrapper to many components that are expecting children
- Updated tsconfig and babelrc with latest expected plugins

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
     [
-      "module-resolver",
+      "@emotion",
       {
-        "alias": {
-          "src": "./src"
-        },
-        "extensions": [".js", ".jsx", ".d.ts", ".ts", ".tsx"],
-        "root": ["."]
+        "autoLabel": "dev-only",
```
Comment (caravinci) [inline .babelrc:11]: useful new feature :)

File: src/AppBar/styledComponents.tsx [hunk 2]
```diff
@@ -31,14 +31,11 @@ export const MenuItemContent = styled(MenuItemTypography)`
...
-  {
-    active: boolean;
-  }
->('div', {
+export const MenuItemContainer = styled('div', {
   shouldForwardProp: (propName) => propName !== 'active',
-})`
+})<{
+  active: boolean;
+}>`
```
Comment (caravinci) [inline src/AppBar/styledComponents.tsx:38]: this syntax is much easier to read 👏

File: src/Pipe/index.tsx [hunk 3]
```diff
@@ -55,9 +55,9 @@ const PipeItem = styled<'div', { fill: keyof typeof defaultTheme.colors }>('div'
   }
 `;
 
-const Pipe: React.ComponentType<{}> & { Item: typeof PipeItem } = ({ children }) => (
-  <PipeContainer>{children}</PipeContainer>
-);
+const Pipe: React.ComponentType<React.PropsWithChildren<{}>> & { Item: typeof PipeItem } = ({
```
Comment (caravinci) [inline src/Pipe/index.tsx:58]: while not relevant to the PR, it is not 100% clear to me what this intersection type works out as, or why it's needed.

File: src/Progress/index.tsx [hunk 4]
```diff
@@ -163,8 +163,8 @@ export const ProgressItem = ({
   </div>
 );
 
-export const Progress: React.ComponentType<{}> & { Item: typeof ProgressItem } = ({ children }) => (
-  <ProgressSection>{children}</ProgressSection>
-);
+export const Progress: React.ComponentType<React.PropsWithChildren<{}>> & {
+  Item: typeof ProgressItem;
```
Comment (caravinci) [inline src/Progress/index.tsx:167]: Same here (not PR related, just a general comment)
...are we extending React Component types to also have an `Item` property? 
seems odd and unidiomatic, but there's probably a reasoning that could just used being documented and spelled out for other devs + users.

File: src/notifications/Notification/styledComponents.tsx [hunk 5]
```diff
@@ -28,7 +28,13 @@ const getBackgroundColor = ({ theme, variant }: { theme?: any; variant: Notifica
     [NOTIFICATION_VARIANTS.WARNING]: theme.colors.warning_4,
     [NOTIFICATION_VARIANTS.ERROR]: theme.colors.error_4,
   }[variant]);
-export const getBorderColor = ({ theme, variant }: { theme?: any; variant: NotificationVariant }) =>
+export const getBorderColor = ({
+  theme,
+  variant,
+}: {
+  theme?: any;
```
Comment (caravinci) [inline src/notifications/Notification/styledComponents.tsx:35]: this type could use a tighter assertion

File: tsconfig.json [hunk 6]
```diff
@@ -21,7 +22,8 @@
     "declaration": true,
     "paths": {
       "src/*": ["./src/*"]
-    }
+    },
+    "types": ["@emotion/react/types/css-prop"]
```
Comment (caravinci) [inline tsconfig.json:26]: I'm surprised this was necessary, based on Emotion's documentation: 
https://emotion.sh/docs/typescript#emotionreact

This line makes part of the old way of adding the `css prop`, and in that case, I'd have thought a `global.d.ts` insertion would've been a better place for it.
https://emotion.sh/docs/typescript#with-the-old-jsx-transform

**Review by caravinci — APPROVED** [review]
awesome!! 🎉 
Just left a couple comments with questions and thoughts.

File: src/notifications/Notification/styledComponents.tsx (hunk 5 above)
Comment (samrichca) [reply src/notifications/Notification/styledComponents.tsx:35]: can likely use uikit's theme type here

File: tsconfig.json (hunk 6 above)
Comment (samrichca) [reply tsconfig.json:26]: good point, I think I added this pre-emotion 11 upgrade. might not be necessary to carry it forward

**Pull request merged**